docs/source/developers/cpp/building.rst

```diff
@@ -145,45 +238,34 @@ Minimal release build (1GB of RAM for building or more recommended):
 
 .. code-block:: shell
 
-   git clone https://github.com/apache/arrow.git
-   cd arrow/cpp
-   mkdir release
-   cd release
-   cmake ..
-   make
+   $ mkdir build-release
+   $ cd build-release
+   $ cmake ..
+   $ make -j8       # if you have 8 CPU cores, otherwise adjust
 
 Minimal debug build with unit tests (4GB of RAM for building or more recommended):
 
 .. code-block:: shell
 
-   git clone https://github.com/apache/arrow.git
-   cd arrow
-   git submodule update --init --recursive
-   export ARROW_TEST_DATA=$PWD/testing/data
-   cd cpp
-   mkdir debug
-   cd debug
-   cmake -DCMAKE_BUILD_TYPE=Debug -DARROW_BUILD_TESTS=ON ..
-   make unittest
+   $ git submodule update --init --recursive
```

Comment:
When giving tutorials, I have noticed that many beginners don't necessarily know the meaning of `$` as signaling "this is on the prompt" (I also think that not all platforms / consoles actually use this symbol?) So I have been actively removing it from instructions (because otherwise people naively copy this line from the docs, and get an error about $). 

If we could make the `$` non-copy-pastable, that would be a nice combination I think.

Reply:
In https://github.com/pydata/pydata-sphinx-theme/issues/41 I listed some options around this. It seems sphinx now does this out of the box, but only for >>> python prompts. https://github.com/sbrunner/sphinx-prompt might be the closest thing.

(but to be clear I don't mean this as a suggestion that needs to be done in this PR)

Reply:
Hmm, I didn't know this confused beginners. I do think it is a nice cue that something is happening at the command prompt, and it also allows to distinguish commands and their output.

How many people building Arrow C++ would be confused about this, though?

Reply:
I think using something like `sphinx-prompt` + having an icon to copy the content of the code box would be great! Something like [sphinx-copybutton](https://sphinx-copybutton.readthedocs.io/en/latest/). Could be a beginner ticket for the docs to add/change this in developers section =)

How many ppl would be confused? Well, not much I would say. But if we could make life easier even for one that is not yet used to the terminal and needs to build Arrow C++ that could be worth it?

Reply:
We can definitely add useful extensions. Perhaps open a JIRA?

Reply:
Created [ARROW-14963](https://issues.apache.org/jira/browse/ARROW-14963).

Reply:
Apparently, sphinx actually already supports having a `$` not copy-pastable out of the box (see eg https://pydata-sphinx-theme.readthedocs.io/en/latest/contributing.html#set-up-a-basic-development-environment), I _think_ it's a matter of specifying the correct highlighting language for the code blocks (`console` was used in the case of the linked example)